R/Spectra.R

```diff
@@ -1161,21 +1169,25 @@ setMethod("filterRt", "Spectra",
 
 #' @rdname Spectra
 #'
-#' @exportMethod removePeaks
-setMethod("removePeaks", "Spectra",
-          function(object, t = "min", msLevel. = unique(msLevel(object))) {
-              if (!is.numeric(t) & t != "min")
-                  stop("Argument 't' has to be either numeric of 'min'.")
-              if (!is.numeric(msLevel.))
-                  stop("'msLevel.' must be numeric.")
-              object <- addProcessing(object, .remove_peaks, t = t,
-                                      msLevel = msLevel.)
-              object@processing <- .logging(object@processing,
-                                            "Signal <= ", t, " in MS level(s) ",
-                                            paste0(msLevel., collapse = ", "),
-                                            " set to 0")
-              object
-          })
+#' @exportMethod bin
+setMethod("bin", "Spectra", function(object, binSize = 1L, breaks = NULL,
+                                     msLevel. = unique(msLevel(object))) {
```

Comment:
I wouldn't know where to store that parameter since the operation is added to the processing queue. Also what happens if the user chooses different `f` for different operations? I'd rather add the `f` parameter to `mz`, `intensity`, `peaks` methods instead.

Reply:
Sry, I meant just to add it to all functions that use `.peaksapply` (as this one to determine `mzr`). But maybe this is over-engineering and nobody ever use that. And you can already apply individual splitting by setting `f` in `applyProcessing` (which is more important).